progenitor-impl/src/method.rs

```diff
@@ -386,7 +393,9 @@ impl Generator {
                 // enum; the generated client method would check for the
                 // content type of the response just as it currently examines
                 // the status code.
-                let typ = if let Some(mt) =
+                let typ = if dropshot_websocket {
+                    OperationResponseType::Upgrade
```

Comment:
Is the Content-type meaningless for the response from one of these endpoints?
https://stackoverflow.com/a/39105228/1371209

I assume it will never be `application/json` (is that true?)

Could we do
```rust
let typ = if let Some(mt) = response.content.get("application/json") {
    // ...
} else if !dropshot_websocket {
    OperationResponseType::Raw
} else {
    OperationResponseType::Upgrade
}
```

My preference (when it doesn't suck!) is to have the typical case first. And can we please update the comment above to cover the new case as well?

Reply:
correct re: it not being application/json - the schema emitted from the API generated by dropshot looks like
```json
"/instance/serial": {
    "get": {
        "operationId": "instance_serial",
        "responses": { "default": { "description": "", "content": { "*/*": { "schema": {} } } } },
        "x-dropshot-websocket": {}
    }
}
```
so `*/*` is what we end up calling it.